### PR TITLE
Biconnectivity

### PR DESCRIPTION
**Pull request by LucaStaus**
Adds functions related to biconnectivity and 2-edge-connectivity.

### CODE REVIEW

File: src/main/kotlin/sipura/alg/Connectivity.kt [hunk 1]
```diff
@@ -163,11 +170,119 @@ object Connectivity {
...
+     *
+     * Runtime: O(n + m)
+     * @return A [Pair] containing a set of all cut-vertices and a set of all bridge-edges.
+     */
+    fun <V> cutVerticesAndBridgeEdges(g: SimpleGraph<V>): Pair<Set<V>, Set<Pair<V, V>>> {
+        val cutVertices = HashSet<V>()
+        val bridgeEdges = HashSet<Pair<V, V>>()
+        val remaining = g.V.toHashSet()
+        // make sure we visit all connected components
+        while (remaining.isNotEmpty()) {
```
Comment (moritz-gross) [inline src/main/kotlin/sipura/alg/Connectivity.kt]: can you add to the docs with algorithm is used?

Comment (moritz-gross) [reply src/main/kotlin/sipura/alg/Connectivity.kt]: also, are you sure we can't reuse some other parts of the library here?

Comment (LucaStaus) [reply src/main/kotlin/sipura/alg/Connectivity.kt]: what do you think I could reuse?

File: src/main/kotlin/sipura/alg/Connectivity.kt [hunk 2]
```diff
@@ -187,6 +187,9 @@ object Connectivity {
      * A vertex v is a cut-vertex if the graph has more connected components after v is removed from it.
      * An edge (v1,v2) is a bridge-edge if the graph has more connected components after (v1,v2) is removed from it.
      *
+     * Uses an algorithm that calculates a dfs-tree and a chain decomposition based on that tree. The cut-vertices
+     * and bridge-edges can then easily be determined from the chain decomposition.
+     *
```
Comment (moritz-gross) [inline src/main/kotlin/sipura/alg/Connectivity.kt:192]: I haven't heard of "chain decompositions" so far.
The first results I found were https://en.wikipedia.org/wiki/Tree_decomposition and http://ktiml.mff.cuni.cz/~gregor/hypercube/lecture17.pdf, which both don't seem to fit.

Maybe add a reference to what you mean?

**Review by moritz-gross — APPROVED** [review]

**Pull request merged**